src/safeposix/syscalls/fs_calls.rs

```diff
@@ -3058,52 +3146,126 @@ impl Cage {
         return dupfd;
     }
 
-    //------------------------------------CLOSE SYSCALL------------------------------------
-
+    /// ## ------------------CLOSE SYSCALL------------------
+    /// ### Description
+    ///
+    /// The `close_syscall()` is used to close a file descriptor, so that it no
+    /// longer refers to any file and may be reused. This is an essential
+    /// operation for managing resources, as file descriptors are a limited
+    /// resource in any operating system.
+    ///
+    /// ### Function Arguments
+    ///
+    /// The `close_syscall()` receives one argument:
+    /// * `fd` - This argument refers to the file descriptor to be closed.
+    ///
+    /// ### Returns
+    ///
+    /// Upon successful completion of this call, it returns 0. Otherwise, it
+    /// returns error number set with the appropriate error code.
+    ///
+    /// ### Errors
+    ///
+    /// * The syscall returns errors which occur in its helper functions.
+    ///
+    /// ### Panics
+    ///
+    /// * There are no such panics which happen inside the function.
```

Comment:
this function indirectly panics when the helper panics - adding that would be good I guess